src/helpers/createTimeoutController.ts

```diff
@@ -12,22 +30,28 @@ function createTimeoutController(timeout: WaitOptions['timeout']) {
       return new Promise<void>((resolve, reject) => {
         timeoutController.timedOut = false
         timeoutController.onTimeout(resolve)
-
         if (timeout) {
           timeoutId = setTimeout(() => {
             timeoutController.timedOut = true
             timeoutCallbacks.forEach((callback) => callback())
             resolve()
           }, timeout)
         }
+        if (jestFakeTimersAreEnabled() && allowFakeTimers) {
```

Comment:
I have to move  this one out of the timeout, otherwise the time will not move forward
for test 'should not reject when waiting for next update if timeout has been disabled'